test/components/util/util.go

```diff
@@ -182,3 +192,79 @@ func filterNonControllerPods(pods *corev1.PodList) *corev1.PodList {
 	pods.Items = filteredPods
 	return pods
 }
+
+type PortForwardInfo struct {
+	ReadyChan chan struct{}
+	StopChan  chan struct{}
+	// Port forwarding only works with a pod. Even if we ask user to provide the service name, we
+	// still have to figure out the pod name and select one of the pod the service is pointing at.
+	// Instead we can rely on the fact that prometheus pods are started by statefulset and will
+	// always have consistent naming.
+	PodName       string
+	Namespace     string
+	PodPort       int
+	LocalPort     int
+	PortForwarder *portforward.PortForwarder
+}
+
+func (p *PortForwardInfo) CloseChan() {
+	close(p.StopChan)
+	close(p.ReadyChan)
+}
+
+func (p *PortForwardInfo) PortForward(t *testing.T) {
```

Comment:
Won't it be easier if we make PortForward non-blocking? It could return both channels and the result of `GetPorts()`.

Reply:
I am not sure if there is a way to do that. But the API provided by go-client is blocking.

Reply:
If we run `forwarder.ForwardPorts()` in goroutine, then it won't block. Let me see if I can implement my idea.

Reply:
why not run the whole method as goroutine then?

Reply:
To have simpler API for the caller for example.